xlators/features/gfid-access/src/gfid-access.c

```diff
@@ -571,7 +571,7 @@ ga_heal_entry(call_frame_t *frame, xlator_t *this, loc_t *loc, data_t *data,
 
     loc_wipe(&tmp_loc);
 
-    if (xdata)
+    if ((ret == 0) && xdata)
```

Comment:
I think it will not work, in case of error after line No. 555 dictionary will not be free.

Reply:
updated the changes.